github_api/prs.py

```diff
@@ -122,14 +122,52 @@ def close_pr(api, urn, pr):
     return api("patch", path, json=data)
 
 
-def get_pr_last_updated(pr_data):
+def get_events(api, pr_owner, pr_repo):
+    """
+    a helper for getting the github events on a given repo... useful for
+    finding out the last push on a repo.
+    """
+    # TODO: this only gets us the latest 90 events, should we do more?
+    # i.e. can someone do 90 events on a repo in 30 seconds?
+    events = []
+    for i in range(1, 4):
+        path = "/repos/{pr_owner}/{pr_repo}/events?page={page}".format(
+                pr_owner=pr_owner, pr_repo=pr_repo, page=i)
+        events += api("get", path, json={})
+
+    return events
+
+
+def get_pr_last_updated(api, pr_data):
     """ a helper for finding the utc datetime of the last pr branch
     modifications """
-    repo = pr_data["head"]["repo"]
-    if repo:
-        return arrow.get(repo["pushed_at"])
-    else:
-        return None
+
+    pr_ref = pr_data["head"]["ref"]
+    pr_repo = pr_data["head"]["repo"]["name"]
+    pr_owner = pr_data["user"]["login"]
+
+    events = get_events(api, pr_owner, pr_repo)
+    events = filter(lambda e: e["type"] == "PushEvent", events)
+
+    # Gives the full ref name "ref/heads/my_branch_name", but we just
+    # want my_branch_name, so isolate it...
+    def ref_name(e):
+        '/'.join(e["payload"]["ref"].split("/")[3:]) == pr_ref
+
+    events = list(filter(ref_name, events))
+
+    if len(events) == 0:
+        # if we can't get good data, fall back to repo push time
+        repo = pr_data["head"]["repo"]
+        if repo:
+            return arrow.get(repo["pushed_at"])
```

Comment:
It didn't work. A number of PRs got insta-closed and one got insta-merged before the threshold increase.